data/LogicHelpers.json

```diff
@@ -61,11 +60,11 @@
             or (for_age == adult and (Bow or Hookshot))
             or (for_age == both and (Slingshot or Boomerang) and (Bow or Hookshot))
             or (for_age == either and (Slingshot or Boomerang or Bow or Hookshot))",
-    "can_bonk_tree": "deadly_bonks != 'ohko' or Fairy or can_use(Nayrus_Love)",
-    "can_bonk_crate": "deadly_bonks != 'ohko' or Fairy or can_use(Nayrus_Love) or can_blast_or_smash",
-    "can_bonk_underwater_crate": "deadly_bonks != 'ohko' or Fairy or can_use(Nayrus_Love)",
-    "can_bonk_heated_crate": "(deadly_bonks != 'ohko' or (Fairy and (can_use(Goron_Tunic) or damage_multiplier != 'ohko')) or can_use(Nayrus_Love) or can_blast_or_smash)",
-
+    "can_bonk": "deadly_bonks != 'ohko' or Fairy or can_use(Nayrus_Love)",
+    "can_break_crate": "can_bonk or can_blast_or_smash",
+    "can_break_heated_crate": "deadly_bonks != 'ohko' or (Fairy and (can_use(Goron_Tunic) or damage_multiplier != 'ohko')) or can_use(Nayrus_Love) or can_blast_or_smash",
```

Comment:
So from my understanding, this line is:

Can bonk  without dying, or
???, or
Have NL to bonk safely, or
Can bomb or hammer crate

What exactly is the 2nd condition? I don't see what having a fairy and either tunic or non-OHKO would do, normally it would be "fairy or non-ohko" for stuff like this

Reply:
If you're talking about can_break_heated_crate? The only things I did was rename the helper and remove an unnecessary set of parentheses around the whole thing. It's otherwise unchanged from main branch.

Reply:
Mracsys felt that forced deaths in a heated area should require Goron Tunic to put out the flames immediately. If I recall correctly, actually, I had to make a change to the logic for the freestandings in crater b/c the fall damage to get down there would have the same problem. The trek to Fire Temple already required Tunic anyway, but now with those freestandings... I had to add similar logic there to be consistent with Mracsys' changes that he made for bonko.

Something that I didn't think until now is that perhaps I should have proposed that not requiring the Goron Tunic to put out the flames (just hold any direction on the joystick) could be folded into the fewer tunic requirements trick. Discussion that I guess I forgot to have.

Reply:
Ah I see, so 2nd condition is a forced death, gotcha.